x-pack/plugins/ingest_pipelines/server/routes/api/create.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+
+import { Pipeline } from '../../../common/types';
+import { API_BASE_PATH } from '../../../common/constants';
+import { RouteDependencies } from '../../types';
+
+const bodySchema = schema.object({
+  name: schema.string(),
+  description: schema.string(),
+  processors: schema.arrayOf(schema.recordOf(schema.string(), schema.any())),
+  version: schema.maybe(schema.number()),
+  onFailure: schema.maybe(schema.arrayOf(schema.recordOf(schema.string(), schema.any()))),
+});
+
+export const registerCreateRoute = ({
+  router,
+  license,
+  lib: { isEsError },
+}: RouteDependencies): void => {
+  router.put(
```

Comment:
To follow REST principles it would be better to use `POST` to create a resource. Tiny change but it goes a long way 😄

Reply:
Good point. I can change this in a follow-up PR. I think I may have used 	`put` here originally to align with the underlying ES API.

Reply:
Change made via https://github.com/elastic/kibana/pull/63017/commits/6bba3dacf16071f2ce0ac5519bfc2a38e684704f